src/crsAdapter/MerlinAdapter.js

```diff
@@ -323,26 +331,107 @@ class MerlinAdapter {
             emptyService.KindOfService = CONFIG.crs.serviceTypes.extras;
             emptyService.Service = hotelName;
             emptyService.FromDate = pickUpDate.isValid() ? pickUpDate.format(CONFIG.crs.dateFormat) : service.pickUpDate;
-            emptyService.EndDate = calculatedDropOffDate;
+            emptyService.EndDate = dropOffDate.isValid() ? dropOffDate.format(CONFIG.crs.dateFormat) : service.dropOffDate;
         }
 
-        xml.Remarks = [xml.Remarks, reduceExtrasList(service.extras), reduceHotelDataToRemarkString(service)].filter(Boolean).join(',') || void 0;
-    };
+        xml.Remarks = [xml.Remarks, reduceHotelDataToRemarkString(service)].filter(Boolean).join(',') || void 0;
+    }
 
     /**
      * @private
      * @param service object
      * @param xmlService object
+     * @param xml object
      */
-    assignHotelServiceFromAdapterObjectToXmlObject(service, xmlService) {
+    assignHotelServiceFromAdapterObjectToXmlObject(service, xmlService, xml) {
         let dateFrom = moment(service.dateFrom, this.options.useDateFormat);
         let dateTo = moment(service.dateTo, this.options.useDateFormat);
+        let travellerAssociation = xmlService.TravellerAllocation || '';
 
         xmlService.KindOfService = CONFIG.crs.serviceTypes.hotel;
         xmlService.Service = service.destination;
         xmlService.Accommodation = [service.roomCode, service.mealCode].filter(Boolean).join(' ');
+        xmlService.StealBoarding = service.roomQuantity;
+        xmlService.Occupancy = service.roomOccupancy;
         xmlService.FromDate = dateFrom.isValid() ? dateFrom.format(CONFIG.crs.dateFormat) : service.dateFrom;
         xmlService.EndDate = dateTo.isValid() ? dateTo.format(CONFIG.crs.dateFormat) : service.dateTo;
+        xmlService.TravellerAllocation = '1' + ((service.roomOccupancy > 1) ? '-' + service.roomOccupancy : '');
+
+        let startLineNumber = parseInt(travellerAssociation.substr(0, 1) || 0, 10);
+        let endLineNumber = parseInt(travellerAssociation.substr(-1) || 0, 10);
+
+        if (!startLineNumber) return;
```

Comment:
Ever since the [goto fail](https://www.imperialviolet.org/2014/02/22/applebug.html) disaster happened, I'd rather use `if`s with curly braces.

Reply:
I see, therefore we use this variant only when a "return" follows immediately. Then put the "return" not in the next line instead append it directly to the if condition on the same line.